design-system/Header/Header.tsx

```diff
@@ -0,0 +1,247 @@
+   */
+  titleContainerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Optional inner header wrapper style override.
+   */
+  style?: StyleProp<ViewStyle>;
+  /**
+   * Optional outer header container style override.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Background color
+   */
+  backgroundColor?: string;
+  /**
+   * Title text to display if not using `tx` or nested components.
+   */
+  title?: ITextProps["text"];
+  /**
+   * Title text which is looked up via i18n.
+   */
+  titleTx?: ITextProps["tx"];
+  /**
+   * Optional options to pass to i18n. Useful for interpolation
+   * as well as explicitly setting locale or translation fallbacks.
+   */
+  titleTxOptions?: ITextProps["txOptions"];
+  /**
+   * Icon that should appear on the left.
+   * Can be used with `onLeftPress`.
+   */
+  leftIcon?: IPicto;
+  /**
+   * An optional tint color for the left icon
+   */
+  leftIconColor?: string;
+  /**
+   * Left action text to display if not using `leftTx`.
+   * Can be used with `onLeftPress`. Overrides `leftIcon`.
+   */
+  leftText?: ITextProps["text"];
+  /**
+   * Left action text text which is looked up via i18n.
+   * Can be used with `onLeftPress`. Overrides `leftIcon`.
+   */
+  leftTx?: ITextProps["tx"];
+  /**
+   * Left action custom ReactElement if the built in action props don't suffice.
+   * Overrides `leftIcon`, `leftTx` and `leftText`.
+   */
+  LeftActionComponent?: ReactElement;
+  /**
+   * Optional options to pass to i18n. Useful for interpolation
+   * as well as explicitly setting locale or translation fallbacks.
+   */
+  leftTxOptions?: ITextProps["txOptions"];
+  /**
+   * What happens when you press the left icon or text action.
+   */
+  onLeftPress?: TouchableOpacityProps["onPress"];
+  /**
+   * Icon that should appear on the right.
+   * Can be used with `onRightPress`.
+   */
+  rightIcon?: IPicto;
+  /**
+   * An optional tint color for the right icon
+   */
+  rightIconColor?: string;
+  /**
+   * Right action text to display if not using `rightTx`.
+   * Can be used with `onRightPress`. Overrides `rightIcon`.
+   */
+  rightText?: ITextProps["text"];
+  /**
+   * Right action text text which is looked up via i18n.
+   * Can be used with `onRightPress`. Overrides `rightIcon`.
+   */
+  rightTx?: ITextProps["tx"];
+  /**
+   * Right action custom ReactElement if the built in action props don't suffice.
+   * Overrides `rightIcon`, `rightTx` and `rightText`.
+   */
+  RightActionComponent?: ReactElement;
+  /**
+   * Optional options to pass to i18n. Useful for interpolation
+   * as well as explicitly setting locale or translation fallbacks.
+   */
+  rightTxOptions?: ITextProps["txOptions"];
+  /**
+   * What happens when you press the right icon or text action.
+   */
+  onRightPress?: ITouchableOpacityProps["onPress"];
+  /**
+   * Override the default edges for the safe area.
+   */
+  safeAreaEdges?: ExtendedEdge[];
+};
+
+export function Header(props: HeaderProps) {
```

Comment:
I feel like it's always nice to see a couple of demo usages of a component with a bunch of props in the comment (since we don't have storybook setup yet). 

Not blocker but could you add if you like that idea?

Reply:
@technoplato Yes! Good idea, I've started doing some "Demo" component for some of them. I really don't like Storybook, so much overkill IMO.

But happy to do a 80/20 solution for now to help

Reply:
80/20 FTW